sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/ShareFileClient.java

```diff
@@ -1642,8 +2119,94 @@ public ShareFileUploadInfo upload(InputStream data, long length, ParallelTransfe
      */
     public Response<ShareFileUploadInfo> uploadWithResponse(ShareFileUploadOptions options,
         Duration timeout, Context context) {
-        return StorageImplUtils.blockWithOptionalTimeout(
-            shareFileAsyncClient.uploadWithResponse(options, context), timeout);
+        StorageImplUtils.assertNotNull("options", options);
```

Comment:
Suggestion, since Sync Stack doesn't need to all happen in one PR, as each API can migrate separately due to its design. Let's keep the upload and download APIs sync-over-async in this PR and handle them as a separate change. Their logic is much more complex than the simple REST API calls most other APIs are doing and warrant their own specific PR without the noise of the large changes elsewhere in this PR.